### PR TITLE
feature: warn when imported csv contains duplicate rows

### PR DESCRIPTION
**Pull request by buckhalt**
This pull request adds a user-facing warning when a CSV asset being imported contains duplicate rows, and includes the duplicate count in the warning dialog. It also introduces supporting infrastructure for this warning, including logic to detect duplicates and new tests to verify the behavior.

**User-facing improvements:**

* Show a warning dialog if a CSV asset contains duplicate rows when importing, including the file name and number of duplicates. This helps users be aware of data quality issues before using the asset. [[1]](diffhunk://#diff-1ecb0aa20b28e5ba8ae10e5e836fa8fb40799961d4ba2ea5d99b830cd2b4467aL52-R64) [[2]](diffhunk://#diff-a8c897679e7a488898e2810722daeab5e47e5446947235640791816995ce7b0aR73-R88)

**Validation logic updates:**

* Add a `countDuplicateRows` helper and update `validateNetwork` in `assetTools.ts` to count duplicate rows in CSV files; the validation result now includes `duplicateCount`. [[1]](diffhunk://#diff-cb04285ade61cf93ae6930e0d719ff572b593c5ac914870ae1a3a56de6a665feL93-R115) [[2]](diffhunk://#diff-cb04285ade61cf93ae6930e0d719ff572b593c5ac914870ae1a3a56de6a665feR126) [[3]](diffhunk://#diff-cb04285ade61cf93ae6930e0d719ff572b593c5ac914870ae1a3a56de6a665feL132-R166)

**Testing improvements:**

* Add tests for the new duplicate row warning behavior in `assetManifest.test.ts`, including both the presence and absence of duplicates. Mocks are added for relevant utility functions. [[1]](diffhunk://#diff-41d42f7bb3533c96a4200dea3bd30a19125086d515a06144d00a134c95369e78R1-R32) [[2]](diffhunk://#diff-41d42f7bb3533c96a4200dea3bd30a19125086d515a06144d00a134c95369e78R71-R103)

**Dialog infrastructure:**

* Add a new `duplicateRowsWarningDialog` action creator in `dialogs.tsx` for consistent warning dialog display.

**Imports and type updates:**

* Update imports and types to support the new validation result and dialog action.

### CODE REVIEW

File: apps/architect-vite/src/ducks/modules/protocol/assetManifest.ts [hunk 1]
```diff
@@ -49,11 +53,15 @@ export const importAssetAsync = createAsyncThunk(
...
 			// Validate asset
-			await validateAsset(file).catch((error) => {
+			const validationResult = await validateAsset(file).catch((error) => {
 				dispatch(invalidAssetErrorDialog(error, name));
 				throw error;
 			});
 
+			if (validationResult.duplicateCount > 0) {
+				dispatch(duplicateRowsWarningDialog(name, validationResult.duplicateCount));
+			}
```
[the remark this answers is not in the document]
Comment (buckhalt) [reply apps/architect-vite/src/ducks/modules/protocol/assetManifest.ts:63]: [961cb1f](https://github.com/complexdatacollective/network-canvas-monorepo/pull/453/commits/961cb1f896631faf208e9b9e71da5cc07fcaf994)
